### PR TITLE
Adjust element segment encoding to match the bulk-memory proposal

### PR DESCRIPTION
**Pull request by gahaas**
Due to recent changes in the bulk-memory proposal (https://github.com/WebAssembly/bulk-memory-operations/pull/109), the element segment encoding here does not match the one in the bulk-memory proposal anymore. Concretely, element segments with flag 0x02 have an additional 0x00 byte after the offset which is a place holder for a potential element kind field.

This PR does minimal changes to get the reference-types proposal and the bulk-memory proposal in sync again.

### CODE REVIEW

File: interpreter/binary/decode.ml [hunk 1]
```diff
@@ -621,21 +625,23 @@ let code_section s =
...
-    match x.Source.it with
-    | 0l -> x
-    | 2l -> at var s
-    | _ -> error s pos "invalid segment kind"
-  in
-  let offset = const s in
-  let init = dat s in
-  {index; offset; init}
+let segment dat kind s =
+  match vu32 s with
```
[the remark this answers is not in the document]
Comment (gahaas) [reply interpreter/binary/decode.ml]: Done.

**Comment by gahaas** [comment]
I had to adjust some tests to address the comments. That was good though, because these tests were not valid anymore.

**Comment by gahaas** [comment]
I'm still waiting for reviews for the same changes to the bulk proposal, see https://github.com/WebAssembly/bulk-memory-operations/pull/110

**Comment by rossberg** [comment]
Aw, sorry, missed that. Flooded mailbox after vacation. :)

**Pull request merged**